docs/abcd_rooparametrichist_tutorial/rooparametrichist_exercise.md

```diff
@@ -0,0 +1,507 @@
+
+```
+
+<details>
+<summary> Datacard with A,B,C,D regions combined  </summary>
+
+```
+Combination of mPhi1500_Catany_2018_CR_B.txt  mPhi1500_Catany_2018_CR_C.txt  mPhi1500_Catany_2018_CR_D.txt  mPhi1500_Catany_2018_SR.txt
+imax 4 number of bins
+jmax 1 number of processes minus 1
+kmax 2 number of nuisance parameters
+----------------------------------------------------------------------------------------------------------------------------------
+shapes Bkg       ch1       param_ws.root wspace:bkg_B
+shapes data_obs  ch1       param_ws.root wspace:data_obs_B
+shapes mPhi_1500  ch1      param_ws.root wspace:mPhi_1500_B
+shapes Bkg       ch2       param_ws.root wspace:bkg_C
+shapes data_obs  ch2       param_ws.root wspace:data_obs_C
+shapes mPhi_1500  ch2      param_ws.root wspace:mPhi_1500_C
+shapes Bkg       ch3       param_ws.root wspace:bkg_D
+shapes data_obs  ch3       param_ws.root wspace:data_obs_D
+shapes mPhi_1500  ch3      param_ws.root wspace:mPhi_1500_D
+shapes Bkg       ch4       param_ws.root wspace:bkg_A
+shapes data_obs  ch4       param_ws.root wspace:data_obs_A
+shapes mPhi_1500  ch4      param_ws.root wspace:mPhi_1500_A
+----------------------------------------------------------------------------------------------------------------------------------
+bin          ch1    ch2    ch3    ch4  
+observation  -1     -1     -1     -1   
+----------------------------------------------------------------------------------------------------------------------------------
+bin                             ch1       ch1       ch2       ch2       ch3       ch3       ch4       ch4     
+process                         mPhi_1500  Bkg      mPhi_1500  Bkg      mPhi_1500  Bkg      mPhi_1500  Bkg     
+process                         0         1         0         1         0         1         0         1       
+rate                            -1        1         -1        1         -1        1         -1        1       
+----------------------------------------------------------------------------------------------------------------------------------
+BkgRate                 lnN     -         -         -         -         -         -         -         1.05    
+lumi                    lnN     1.016     -         1.016     -         1.016     -         1.016     -
+
+
+```
+</details>
+
+
+
+## Run Fit
+<a id="fit"></a>
+
+From the combined datacard for all the regions, one can run the usual fit diagnostics as follows:
+
+```
+combine -M FitDiagnostics combined_mPhi_1500_2018.txt -m 1500 --saveShapes --saveWithUncertainties --saveNormalizations
+
+```
+
+<details>
+<summary> Results FitDiagnostics  </summary>
+
+```
+ --- FitDiagnostics ---
+Best fit r: 2.4196e-05  -2.4196e-05/+0.276209  (68% CL)
+
+```
+</details>
+
+
+Using the output ```fitDiagnosticsTest.mH1500.root```, one can run the script ```$CMSSW_BASE/src/HiggsAnalysis/CombinedLimit/test/mlfitNormsToText.py``` to get the predictions for the normalizations:
```

Comment:
The output file in this case does not include the mass parameter, its just `fitDiagnosticsTest.root`

Reply:
this is strange, since i get exactly that outfile name running this command: ```combine -M FitDiagnostics combined_mPhi_1500_2018.txt -m 1500 --saveShapes --saveWithUncertainties --saveNormalizations```. I would expect the ```.mH1500.``` part since the mass parameter is specified

Reply:
i have updated, now is consistent with what you also get in terms of filename (was running with an older version)

Reply:
Are you sure? The filenames produced for the files which contain the limit tree (which do always contain the mass parameter, but is not relevant here) do not follow the same output formatting as the fitDiagnostic file which saves the shapes and RooFitResults. The `higgsCombineTest.FitDiagnostics.mH1500.root` file should also be produced but that just contains the limit tree, the `fitDiagnosticsTest.root` file doesn't rely on the mass value and only uses the `-n` argument.